### PR TITLE
Switching networks header component

### PR DESCRIPTION
**Pull request by marcinciarka**
# [[L2] Network (chain) switcher on the page header](https://app.shortcut.com/oazo-apps/story/8350/l2-add-a-component-for-switching-the-networks-on-the-page-header)
  
## Changes 👷‍♀️
- network switching component:
image
image

- smaller changes: chains system names changed, eg. `main` ➡️ `ethereumMainnet`
- two new flags in the chain configs:
```
  testnet: false, // not showing this chain unless its local instance or theres a testnets=1 query param
  enabled: true, // just a toggle
```
- smaller changes: vague `LOCALSTORAGE_KEY` cookie LS key changed to `COOKIE_NAMES_LOCASTORAGE_KEY`
- smaller changes: changed how the `useLocalStorage` hook is working, now the changes are dynamic and better typed
- smaller changes: changed how the `useCustomNetworkParameter` hook is working, now the param is stored in local storage (instead of query param):
```
const [data , setCustomNetwork] = useCustomNetworkParameter()

// {
//   "network": "ethereumMainnet",
//   "id": "1",
//   "hexId": "0x1"
// }
```
- smaller changes: updated `web3-onboard` + `usehooks-ts` package

  
## How to test 🧪
- TBD

### CODE REVIEW

File: components/navigation/NavigationPickNetwork.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  }
+  return false
+}
+
+export function NavigationPickNetwork() {
+  const { switchChains } = useAppContext()
+  const [{ chains: usableChains, settingChain, connectedChain }, setChain] = useSetChain()
+  const currentNetworkName = useNetworkName()
+  const [, setCustomNetwork] = useCustomNetworkParameter()
+  const changeChain = (networkName: string) => () => {
```
Comment (jakubswierczek) [inline components/navigation/NavigationPickNetwork.tsx:29]: wrap with `useCallback`

File: features/web3OnBoard/useBridgeConnection.tsx [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
+import { BridgeConnector } from './BridgeConnector'
 import { useBridgeConnector } from './useBridgeConnector'
 
 export function useBridgeConnection() {
   const createBridgeConnector = useBridgeConnector()
+  const [{ wallet }] = useConnectWallet()
+  const [{ chains }] = useSetChain()
   const { web3Context$ } = useAppContext()
   const [web3Context] = useObservable(web3Context$)
+  const web3NotConnected =
```
Comment (jakubswierczek) [inline features/web3OnBoard/useBridgeConnection.tsx:15]: `useMemo`

File: features/web3OnBoard/useBridgeConnection.tsx [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+    web3Context?.status === 'error' ||
+    web3Context?.status === 'notConnected' ||
+    web3Context?.status === 'connectedReadonly'
+
+  useEffect(() => {
```
Comment (jakubswierczek) [inline features/web3OnBoard/useBridgeConnection.tsx:20]: I prefer to move the logic related to connecting to `useBridgeConnector`. This way, we will have the core logic in one place.

File: features/web3OnBoard/useBridgeConnector.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ import { BridgeConnector } from './BridgeConnector'
 export function useBridgeConnector(): () => Promise<BridgeConnector | undefined> {
   const [, connect, disconnect] = useConnectWallet()
```
Comment (jakubswierczek) [inline features/web3OnBoard/useBridgeConnector.tsx:9]: We can add a `wallet` here. Then, in line 42, check if the wallet exists and create a new bridge connector if it does.

File: features/web3OnBoard/useWeb3OnBoardConnection.tsx [hunk 5]
```diff
@@ -30,5 +35,11 @@ export function useWeb3OnBoardConnection({ walletConnect }: { walletConnect: boo
     }
   }, [networkConnect, connect, walletConnect])
 
+  useEffect(() => {
```
Comment (jakubswierczek) [inline features/web3OnBoard/useWeb3OnBoardConnection.tsx:38]: This hook works as a chain of callbacks, so effects should not be used. What is the difference between this effect and the `changeNetwork` callback in `useBridgeConnector`?

File: helpers/networkIconMap.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const networkIconMap: Record<string, string> = {
+  ethereumMainnet: '',
+  ethereumHardhat: '👷 ',
+  ethereumGoerli: '🌲 ',
+  arbitrumMainnet: '',
+  arbitrumGoerli: '🌲 ',
+  avalancheMainnet: '',
+  optimismMainnet: '',
```
Comment (jakubswierczek) [inline helpers/networkIconMap.ts:8]: As we have `arbitrumGoerli`, we should also add `optimismGoerli`

**Review by jakubswierczek — COMMENTED** [review]
👍

File: components/navigation/NavigationPickNetwork.tsx (hunk 1 above)
Comment (marcinciarka) [reply components/navigation/NavigationPickNetwork.tsx:29]: Done.

File: features/web3OnBoard/useBridgeConnection.tsx (hunk 2 above)
Comment (marcinciarka) [reply features/web3OnBoard/useBridgeConnection.tsx:15]: Done.

File: features/web3OnBoard/useBridgeConnection.tsx (hunk 3 above)
Comment (marcinciarka) [reply features/web3OnBoard/useBridgeConnection.tsx:20]: Done.

**Pull request merged**